File: scala/src/test/scala/io/github/configur8/ExposeModeTest.scala
package io.github.configur8

import org.scalatest.{FunSpec, Matchers}

class ExposeModeTest extends FunSpec with Matchers {

  describe("Expose mode") {
    it("Public") {
      ExposeMode.Public.display("input") shouldBe "input"
    }

    it("Private conceals the value with '*' characters") {
      ExposeMode.Private.display("input") shouldBe "*********"
    }
  }
}

File: java/src/test/java/io/github/configur8/ExposeModeTest.java
package io.github.configur8;

import org.junit.Test;

import static io.github.configur8.ExposeMode.*;
import static io.github.configur8.ExposeMode.Private;
import static org.hamcrest.Matchers.equalTo;
import static org.junit.Assert.assertThat;

public class ExposeModeTest {

    @Test
    public void publicDisplaysValue() throws Exception {
        assertThat(Public.display("hello"), equalTo("hello"));
    }

    @Test
    public void privateMasksValue() throws Exception {
        assertThat(Private.display("hello"), equalTo("*********"));
    }
}
